seckill-query-service/src/main/java/io/servicecomb/poc/demo/seckill/SecKillEventSubscriber.java

```diff
@@ -0,0 +1,42 @@
+package io.servicecomb.poc.demo.seckill;
+
+import io.servicecomb.poc.demo.seckill.event.PromotionEvent;
+import io.servicecomb.poc.demo.seckill.event.PromotionEventType;
+import io.servicecomb.poc.demo.seckill.repositories.PromotionRepository;
+import io.servicecomb.poc.demo.seckill.repositories.SpringBasedCouponEventRepository;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class SecKillEventSubscriber {
+
+  private SpringBasedCouponEventRepository couponEventRepository;
```

Comment:
query service shall not be aware of event store. it doesn't query database for events but user-coupon relationship directly